packages/hoppscotch-common/src/helpers/import-export/import/openapi.ts

```diff
@@ -754,19 +754,21 @@ const parseOpenAPIUrl = (
    **/
 
   if (objectHasProperty(doc, "swagger")) {
-    const host = doc.host?.trim() || "<<baseUrl>>"
-    const basePath = doc.basePath?.trim() || ""
-    return `${host}${basePath}`
+    // TODO: dynamically add doc.host, doc.basePath value as variables in the environment if available. or notify user to add it.
+    // add base url variable to each request
+    const hostVar = "<<baseUrl>>"
+    const basePathVar = doc.basePath ? "<<basePath>>" : ""
+    return `${hostVar}${basePathVar}`
   }
 
   /**
    * OpenAPI V3 has version as a string in the document's openapi property.
    * And host and basePath are in the document's servers property.
    * Relevant v3 reference: https://swagger.io/specification/#server-object
    **/
-  if (objectHasProperty(doc, "servers")) {
-    return doc.servers?.[0]?.url ?? "<<baseUrl>>"
-  }
+  // if (objectHasProperty(doc, "servers")) {
+  // TODO: dynamically add doc.servers[0].url value as variable in the environment if available. or notify user to add it.
+  // }
```

Comment:
```suggestion
if (objectHasProperty(doc, "servers")) {
    // TODO: dynamically add server URL value as variable in the environment if available, or notify user to add it.
    const serverUrl = doc.servers?.[0]?.url

    if (!serverUrl || serverUrl === "./") {
      return "<<baseUrl>>"
    }

    return serverUrl
  }
```

As adding the server URL to environment variables is not currently supported, we can extend the business logic to accommodate specific use cases for flexibility, with a comment noting this for future review.